packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -98,13 +98,16 @@ class BuildWebCommand extends BuildSubCommand {
     );
     argParser.addFlag(
       'omit-type-checks',
-      help: 'Omit type checks in Wasm output.',
+      help: 'Omit type checks in Wasm output.\n'
+          'Reduces code size and improves performance, but may affect runtime correctness. Use with care.',
       negatable: false,
       hide: !featureFlags.isFlutterWebWasmEnabled,
     );
     argParser.addFlag(
       'wasm-opt',
-      help: 'Run wasm-opt on the output wasm module.',
+      help:
+          'Optimize output wasm using the Binaryen (https://github.com/WebAssembly/binaryen) tool.\n'
+          'Increases the build time, but will yield a smaller, faster output.',
```

Comment:
I think this should be opt out. The main goal is to show that wasm is faster, so the default should zipline straight to that goal.

Reply:
Uh. Okay. Funny. I was going to add a `--debuginfo` line for wasmopt.

Maybe we opt-in to wasmopt by default but also add `--debuginfo` so we can debug issues.

Reply:
Maybe we should just make this dependent on debug vs profile vs release. That would be
* debug = no wasm-opt
* profile = wasm-opt but add -g so that we get debug info
* release = wasm-opt and no -g
That seems most consistent with normal expectations, and it means that wasm-opt is opt-out instead of opt-in.

Reply:
That's a GREAT idea. Will do in a follow-up!